pkg/session_recording/recordings.jsx

```diff
@@ -398,6 +491,35 @@
             this.journalctl = null;
         }
 
+        /*
+         * Restarts journalctl.
+         * Assumes journalctl is running.
+         */
+        journalctlRestart() {
+            if(this.journalctlIsRunning()) {
+                this.journalctl.stop();
+                this.recordingMap = {};
+                this.setState({recordingList: []});
+            }
+            this.journalctlStart();
+        }
+
+        /*
+         * Handles Datepicker Since.
+         */
+        handleDateSinceChange(date) {
+            this.setState({dateSince: date});
+            this.journalctlRestart();
```

Comment:
I'm not sure having journalctlRestart here would be correct. First of all, the date doesn't necessarily change, and there might be no update, or it can be queued for later. See https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

I would say, we need to try putting the journalctl restart into componentDidUpdate. However, any associated state change needs to be here.